Task3/scripts/components/component.js

```diff
@@ -0,0 +1,33 @@
+class Component {
+    constructor(options) {
+        this._el = options.element;
+    }
+
+    get element(){
+        return this._el;
+    }
+
+    hide() {
+        this._el.classList.add('js-hidden');
+    }
+
+    show() {
+        this._el.classList.remove('js-hidden');
+    }
+
+    on(eventName, handler) {
+        this._el.addEventListener(eventName, handler);
+    }
+
+    _trigger(eventName, data, options) {
+        options = options || {};
+
+        if (data !== undefined && data !== null) {
```

Comment:
if you don't need to check for 0 you can just check for data (if (data)... )

Then it can be simplified to options.detail = data || 0;